### ISSUE TITLE
e2e flake: TestClusterRestore (data copy job pending timeout)

### ISSUE THREAD

**Issue by hongchaodeng** [coreos/etcd-operator#588]
https://jenkins-etcd-public.prod.coreos.systems/job/etcd-operator/1093/console:
```
--- FAIL: TestClusterRestore (300.41s)
    --- FAIL: TestClusterRestore/restore_cluster_from_backup (0.00s)
        --- FAIL: TestClusterRestore/restore_cluster_from_backup/restore_from_the_same_name_cluster (101.45s)
        	restore_test.go:121: fail to wait pods deleted: still failing after 5 retries
        --- FAIL: TestClusterRestore/restore_cluster_from_backup/restore_from_a_different_name_cluster (300.41s)
        	restore_test.go:126: failed to create 3 members etcd cluster: still failing after 24 retries
        	restore_test.go:121: still failing after 5 retries
```

**Comment by hongchaodeng** [comment]
The first failure was introduced by the new change in #585. Which we didn't merge, so we don't take as flake.

**Comment by hongchaodeng** [comment]
The second failure was due to data copy job wait timeout:
```
time="2017-01-04T06:45:52Z" level=error msg="fail to setup: fail to wait data copy job completed: timed out waiting for the condition
```

It waits for such condition:
```
// PodCompleted returns true if the pod has run to completion, false if the pod has not yet
// reached running state, or an error in any other case.
func PodCompleted(event watch.Event) (bool, error) {
```

Actually our code logic isn't right that it could be "Failed" for PodCompleted to return true. But that isn't about this flake.

Back to the flake, it just means that the pod keeps pending.

**Comment by xiang90** [comment]
> The second failure was due to data copy job wait timeout:

Why the copy job times out? The data should be under 1MB.

**Comment by hongchaodeng** [comment]
> Why the copy job times out? The data should be under 1MB.

It didn't even start. It keeps pending.

**Comment by xiang90** [comment]
But why it keeps pending?

**Comment by hongchaodeng** [comment]
@xiang90 

I actually think our current way to "copying volumes" isn't a clean and simple way to achieve this.
We probably shouldn't copy, but directly mount the volume. The volume management should be up to admin/users.

**Comment by hongchaodeng** [comment]
> But why it keeps pending?

My guess is volume detaching. You can try it out and have a feeling how long this whole thing interacts..

**Comment by hongchaodeng** [comment]
ref: PodFailed check in #590

**Comment by hongchaodeng** [comment]
We encountered this again.

Logs:
```
fail to setup: fail to wait backup copy pod (test-etcd-eg8s2-copyvolume) succeeded: still failing after 12 retries
```

It's definitely not "Failed". I'm 99% sure it's "Pending", sent out #611 for further info.
If so, the pod just ends up waiting for k8s controller to detach volumes and time out after 120s.

**Comment by hongchaodeng** [comment]
Cause:
- The reason why it took so long between deleting previous cluster and receiving a new cluster to restore is that our test is concurrent but operator is blocking. In our test, we have two restore cases to do, each creation of seed member will take 90s. This will likely cause another to fail.

**Comment by hongchaodeng** [comment]
Happened again:

This time it's a little different:
- it successfully created the first cluster, deleted the cluster, and restored the seed.
- it failed to create the third member of the second cluster (restored)
```
restore_test.go:126: failed to create 3 members etcd cluster: still failing after 18 retries
```

**Comment by hongchaodeng** [comment]
fixed via #639
